### PR TITLE
Streams registry

### PR DESCRIPTION
**Pull request by ferjm**

### CODE REVIEW

**Comment by Manishearth** [comment]
Is there a reason the streams registry is only in the gstreamer backend? I'd imagine that the registry would be common to all backends.

There _will_ be client implementations of MediaStream eventually, for example. (since the DOM can write custom media streams)

**Comment by Manishearth** [comment]
_eventually_ we'll probably want to switch this over to an event loop model, but that's only for when we move all of servo-media to its own process, which comes much later, and can be done pretty easily.

**Comment by ferjm** [comment]
> Is there a reason the streams registry is only in the gstreamer backend? I'd imagine that the registry would be common to all backends.

No specific reason. It makes sense to share it among backends. I'll update the patch tomorrow. Thanks!

**Comment by ferjm** [comment]
I've just updated the PR with the suggested change.

File: backends/dummy/lib.rs [hunk 1]
```diff
@@ -36,24 +37,24 @@ impl BackendInit for DummyBackend {
 }
 
 impl Backend for DummyBackend {
-    fn create_audiostream(&self) -> Box<MediaStream> {
-        Box::new(DummyMediaStream)
+    fn create_audiostream(&self) -> MediaStreamId {
+        MediaStreamId::new()
```
Comment (Manishearth) [inline backends/dummy/lib.rs]: Should the dummy backend be registering dummy streams instead? `register_stream(Box::new(DummyStream))`?

**Pull request merged**